service/header/subscription_test.go

```diff
@@ -54,9 +50,33 @@ func TestSubscriber(t *testing.T) {
 	err = p2pSub2.Start(context.Background())
 	require.NoError(t, err)
 
+	done := make(chan bool, 1)
+	go func() {
+		for {
+			if net.Hosts()[1].Peerstore().Peers().Len() == 2 && net.Hosts()[0].Peerstore().Peers().Len() == 2 {
```

Comment:
We can make this even better by using Host's EventBus and waiting for `event.EvtPeerIdentificationCompleted`. Would be cool if you can implement this instead to remove sleeps and this loop entirely.

Reply:
That is a good idea. I think we should open an issue and use this approach where feasible in general (I'm sure there are more places where this could be applied). Might be reasonable to do this in a followup PR later.

Reply:
added it, seems to work nicely.